pkg/mcproducer/mcproducer.go

```diff
@@ -0,0 +1,62 @@
+package mcproducer
+
+import (
+	"bytes"
+	"embed"
+	"encoding/base64"
+	"fmt"
+	"text/template"
+
+	"github.com/google/go-containerregistry/pkg/name"
+)
+
+//go:embed templates
+var templateFS embed.FS
+
+func ProduceMachineConfig(machineConfigName, machineConfigPoolRef, kernelModuleImage, kernelModuleName string) (string, error) {
+	tag, err := name.NewTag(kernelModuleImage)
+	if err != nil {
+		return "", fmt.Errorf("invalid kernelModuleImage %s, input should be repo:tag format: %v", kernelModuleImage, err)
+	}
+
+	templateParams := map[string]interface{}{
+		"Image":                tag.Repository.Name(),
+		"Tag":                  tag.Identifier(),
+		"KernelModule":         kernelModuleName,
+		"MachineConfigPoolRef": machineConfigPoolRef,
+		"MachineConfigName":    machineConfigName,
+	}
+
+	replaceKernelModuleScript, err := generateTemplate(templateFS, "templates/replace-kernel-module.gotmpl", templateParams)
+	if err != nil {
+		return "", err
+	}
+	pullKernelModuleScript, err := generateTemplate(templateFS, "templates/pull-image.gotmpl", templateParams)
+	if err != nil {
+		return "", err
+	}
+	replaceScriptBase64 := base64.StdEncoding.EncodeToString([]byte(replaceKernelModuleScript))
```

Comment:
I think we should instead use [`base64.NewEncoder`](https://pkg.go.dev/encoding/base64#NewEncoder) and use it as the template output - less copies.